src/main/java/rueppellii/backend2/tribes/resource/service/ResourceService.java

```diff
@@ -54,17 +49,68 @@ public void minusGoldAmount(Integer gold, Long kingdomId) throws NoResourceExcep
         saveResource(resource);
     }
 
-    public void plusGoldAmount(Integer gold, Long kingdomId) throws NoResourceException {
-        Resource resource = returnResource(ResourceType.GOLD, kingdomId);
-        resource.setAmount(resource.getAmount() + gold);
-        saveResource(resource);
-    }
-
-    public static List<Resource> starterKit(){
+    public static List<Resource> starterKit(Kingdom kingdom) {
         List<Resource> starterResources = new ArrayList<>();
         for (ResourceType t : ResourceType.values()) {
             starterResources.add(makeResource(t));
         }
+        starterResources.forEach(resource -> resource.setResourcesKingdom(kingdom));
         return starterResources;
     }
+
+    public void updateResources(Kingdom kingdom) {
+        List<Resource> resources = kingdom.getKingdomsResources();
+        for (Resource r : resources) {
+            Integer baseResourcePerMinute = r.getResourcePerMinute();
+            Integer totalBuildingLevelMultiplier = getTotalBuildingLevelMultiplier(kingdom.getKingdomsBuildings(), r);
+            Integer elapsedSeconds = timeService.calculateElapsedSeconds(r.getUpdatedAt());
+            Long remainderSeconds = timeService.calculateRemainder(r.getUpdatedAt());
+            if (r instanceof Gold && calculateGold(baseResourcePerMinute, totalBuildingLevelMultiplier, elapsedSeconds) != 0) {
+                r.setAmount(r.getAmount() + calculateGold(baseResourcePerMinute, totalBuildingLevelMultiplier, elapsedSeconds));
+                r.setUpdatedAt(System.currentTimeMillis() + remainderSeconds);
+            }
+            if (r instanceof Food && calculateFood(kingdom, baseResourcePerMinute, totalBuildingLevelMultiplier, elapsedSeconds) != 0) {
+                r.setAmount(r.getAmount() + calculateFood(kingdom, baseResourcePerMinute, totalBuildingLevelMultiplier, elapsedSeconds));
+                r.setUpdatedAt(System.currentTimeMillis() + remainderSeconds);
+            }
+        }
+        kingdomService.save(kingdom);
+    }
+
+    public Integer getTotalBuildingLevelMultiplier(List<Building> kingdomsBuildings, Resource resource) {
+        String buildingName = null;
+        if (resource instanceof Gold) {
+            buildingName = "MINE";
+        }
+        if (resource instanceof Food) {
+            buildingName = "FARM";
+        }
+        String finalBuildingName = buildingName;
```

Comment:
unnecessary new variable

Reply:
since then this branch is part of my PR (the working part of it, because Archie broke it later) that's why i'm replying: for the lambda expression the variable needs to be final but it can't be declared final because it's changed in the if conditionals